app/scripts/contentscript.js

```diff
@@ -285,3 +287,33 @@ function redirectToPhishingWarning () {
     href: window.location.href,
   })}`
 }
+
+function getSiteName (window) {
+  const document = window.document
+  const siteName = document.querySelector('head > meta[property="og:site_name"]')
+  if (siteName) {
+    return siteName.content
+  }
+
+  return document.title
+}
+
+function getSiteIcon (window) {
+  const document = window.document
+
+  // Use the site's favicon if it exists
+  const shortcutIcon = document.querySelector('head > link[rel="shortcut icon"]')
+  if (shortcutIcon) {
+    return shortcutIcon.href
+  }
+
+  // Search through available icons in no particular order
+  const icons = Array.from(document.querySelectorAll('head > link[rel="icon"]'))
+  for (let i = 0; i < icons.length; i++) {
```

Comment:
Any reason for a `for` loop instead of `.find()`?

Reply:
Nope, I can rewrite it